### PR TITLE
[CLApp] Registering Plane Strain iso+kin plasticity

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
**📝 Description**
That's it. Just registering the plane strain version of the plasticity laws. Some minor optimizations in cls.

@SergioJimenezReyes @rubenzorrilla

### CODE REVIEW

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp [hunk 1]
```diff
@@ -101,6 +101,30 @@ void KratosConstitutiveLawsApplication::Register()
     KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DTrescaMohrCoulomb", mSmallStrainIsotropicPlasticity3DTrescaMohrCoulomb);
     KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticity3DDruckerPragerMohrCoulomb", mSmallStrainIsotropicPlasticity3DDruckerPragerMohrCoulomb);
 
+    KRATOS_REGISTER_CONSTITUTIVE_LAW("SmallStrainIsotropicPlasticityPlaneStrainVonMisesVonMises", mSmallStrainIsotropicPlasticityPlaneStrainVonMisesVonMises);
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:104]: MOAR LAWS!

File: applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/cl_integrators/generic_cl_integrator_kinematic_plasticity.h [hunk 2]
```diff
@@ -279,8 +279,8 @@ class GenericConstitutiveLawIntegratorKinematicPlasticity
         const Vector& rBackStressVector
         )
     {
-        BoundedArrayType deviator = ZeroVector(6);
-        BoundedArrayType h_capa = ZeroVector(6);
+        BoundedArrayType deviator = ZeroVector(VoigtSize);
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/cl_integrators/generic_cl_integrator_kinematic_plasticity.h:282]: :)

File: applications/ConstitutiveLawsApplication/custom_constitutive/small_strains/plasticity/generic_small_strain_isotropic_plasticity.cpp [hunk 3]
```diff
@@ -332,9 +335,12 @@ void GenericSmallStrainIsotropicPlasticity<TConstLawIntegratorType>::FinalizeMat
 
     // Initialize Plastic Parameters
     double uniaxial_stress = 0.0, plastic_denominator = 0.0;
-    BoundedArrayType f_flux = ZeroVector(VoigtSize); // DF/DS
-    BoundedArrayType g_flux = ZeroVector(VoigtSize); // DG/DS
-    BoundedArrayType plastic_strain_increment = ZeroVector(VoigtSize);
+    array_1d<double, VoigtSize> f_flux; // DF/DS
+    array_1d<double, VoigtSize> g_flux; // DG/DS
+    array_1d<double, VoigtSize> plastic_strain_increment;
+    f_flux.clear();
```
Comment (loumalouomega) [inline applications/ConstitutiveLawsApplication/custom_constitutive/small_strains/plasticity/generic_small_strain_isotropic_plasticity.cpp:341]: I always have doubts what is faster to clear or manually assign values

**Review by loumalouomega — APPROVED** [review]

File: applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp (hunk 1 above)
Comment (AlejandroCornejo) [reply applications/ConstitutiveLawsApplication/constitutive_laws_application.cpp:104]: People are asking for them!

**Pull request merged**